articles/advanced-r-concepts/index.md

```diff
@@ -0,0 +1,333 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/advanced-r-concepts/
+title: Advanced Programming Concepts in R
+description:
+author: lalithnarayan-c
+date: 2020-10-15T00:00:00-11:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/advanced-r-concepts/hero.jpg
+    alt:
+---
+In the first article, we have covered the basics of the R programming language. In this article, we will look at some of the advanced aspects of the R programming language. Going through this article will enhance one's ability to implement Object-Oriented Programming(OOP) concepts and write modular code. We will cover the following topics in this article:
+<!--more-->
+
+1. Dataframes
+2. Typecasting
+3. File handling
+4. Working with inbuilt datasets
+5. Conditional Statements
+6. Loop Statements
+
+### Dataframes
+R is a programming language built for statistical analysis of large datasets. Emphasis on handling large datasets is laid via dataframes. Dataframes facilitate the process of storing large datasets in a local variable.
+
+Let's look at an example to define a dataframe. We will describe several vectors storing different data types. The example considered below is a simplified version of a real dataset. Datasets contain columns of different data types.
+
+```r
+vector_1 <- c("a","b","c","d","e")
+vector_2 <- c("A","B","C","D","E")
+vector_3 <- c(1,2,3,4,5)
+vector_4 <- c(2,4,6,8,10)
+vector_5 <- c(TRUE,TRUE,TRUE,FALSE,FALSE)
+
+new_data_frame <- data.frame(vector_1, vector_2, vector_3, vector_4, vector_5)
```

Comment:
As with previous article, this needs comments. You are creating a data frame out of 5 vectors.